x-pack/plugins/upgrade_assistant/__jest__/client_integration/cluster.test.ts

```diff
@@ -0,0 +1,363 @@
+            'model snapshot [%s] for job [%s] supports minimum version [%s] and needs to be at least [%s]',
+          url: 'doc_url',
+          correctiveAction: {
+            type: 'mlSnapshot',
+            snapshotId,
+            jobId,
+          },
+        },
+      ],
+      indices: [],
+    };
+
+    beforeEach(async () => {
+      httpRequestsMockHelpers.setLoadEsDeprecationsResponse(upgradeStatusMockResponse);
+      httpRequestsMockHelpers.setLoadDeprecationLoggingResponse({ isEnabled: true });
+
+      await act(async () => {
+        testBed = await setupClusterPage({ isReadOnlyMode: false });
+      });
+
+      const { actions, component } = testBed;
+
+      component.update();
+
+      // Navigate to the cluster tab
+      await act(async () => {
+        actions.clickTab('cluster');
+      });
+
+      component.update();
+    });
+
+    test('renders deprecations', () => {
+      const { exists } = testBed;
+      expect(exists('clusterTabContent')).toBe(true);
+      expect(exists('deprecationsContainer')).toBe(true);
+    });
+
+    describe('fix ml snapshots button', () => {
+      let flyout: Element | null;
+
+      beforeEach(async () => {
+        const { component, actions, exists, find } = testBed;
+
+        expect(exists('deprecationsContainer')).toBe(true);
+
+        // Open all deprecations
+        actions.clickExpandAll();
+
+        // The data-test-subj is derived from the deprecation message
+        const accordionTestSubj = `depgroup_${upgradeStatusMockResponse.cluster[0].message
+          .split(' ')
+          .join('_')}`;
+
+        await act(async () => {
+          find(`${accordionTestSubj}.fixMlSnapshotsButton`).simulate('click');
+        });
+
+        component.update();
+
+        // We need to read the document "body" as the flyout is added there and not inside
+        // the component DOM tree.
+        flyout = document.body.querySelector('[data-test-subj="fixSnapshotsFlyout"]');
+
+        expect(flyout).not.toBe(null);
+        expect(flyout!.textContent).toContain('Upgrade or delete model snapshot');
+      });
+
+      test('upgrades snapshots', async () => {
+        const { component } = testBed;
+
+        const upgradeButton: HTMLButtonElement | null = flyout!.querySelector(
+          '[data-test-subj="upgradeSnapshotButton"]'
+        );
+
+        httpRequestsMockHelpers.setUpgradeMlSnapshotResponse({
+          nodeId: 'my_node',
+          snapshotId,
+          jobId,
+          status: 'in_progress',
+        });
+
+        await act(async () => {
+          upgradeButton!.click();
+        });
+
+        component.update();
+
+        // First, we expect a POST request to upgrade the snapshot
+        const upgradeRequest = server.requests[server.requests.length - 2];
+        expect(upgradeRequest.method).toBe('POST');
+        expect(upgradeRequest.url).toBe('/api/upgrade_assistant/ml_snapshots');
+        expect(upgradeRequest.status).toEqual(200);
```

Comment:
nit: I don't think we need to check for request status here, since the response is mocked anyway. There are several places in this file where I think we only want to check what we send in the request, but not necessarily want is returned.